File: provisioning/src/main/java/com/ellactron/provissioning/configuration/ServiceConfigure.scala
package com.ellactron.provissioning.configuration

/**
  * Created by ji.wang on 2017-05-09.
  */

import net.tinybrick.security.authentication.ISecurityService
import org.apache.log4j.Logger
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.context.annotation.{Bean, Configuration, PropertySource}
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer

@EnableAutoConfiguration
@Configuration
@PropertySource(value = Array(
  "classpath:config/config.properties"))
class ServiceConfigure {
  val logger = Logger.getLogger(this.getClass())

  @Bean def propertySourcesPlaceholderConfigurer(): PropertySourcesPlaceholderConfigurer = new PropertySourcesPlaceholderConfigurer()
  //@Bean def SecurityService: ISecurityService = new com.ellactron.provissioning.security.SecurityService()
}

File: provisioning/src/main/java/com/ellactron/provissioning/configuration/ServiceConfigure.java
package com.ellactron.provissioning.configuration;

import com.ellactron.provissioning.controllers.UserRegister;
import com.ellactron.provissioning.services.SecurityService;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;

/**
 * Created by ji.wang on 2017-07-24.
 */
@EnableAutoConfiguration
@Configuration
@PropertySource({"classpath:config/config.properties"})
public class ServiceConfigure {
    @Bean
    @ConditionalOnBean(SecurityService.class)
    public UserRegister getUserRegister() {
        return new UserRegister();
    }

    @Bean
    public SecurityService getSecurityService() {
        return new SecurityService();
    }
}


